pkg/agent/metrics/metrics.go

```diff
@@ -46,49 +56,158 @@ var (
 
 // AgentMetrics includes all the metrics of the proxy agent.
 type AgentMetrics struct {
-	latencies *prometheus.HistogramVec
-	failures  *prometheus.CounterVec
+	dialLatencies       *prometheus.HistogramVec
+	serverFailures      *prometheus.CounterVec
+	dialFailures        *prometheus.CounterVec
+	serverConnections   *prometheus.GaugeVec
+	endpointConnections *prometheus.GaugeVec
+	packets             *prometheus.CounterVec
+	streamErrors        *prometheus.CounterVec
 }
 
 // newAgentMetrics create a new AgentMetrics, configured with default metric names.
 func newAgentMetrics() *AgentMetrics {
-	latencies := prometheus.NewHistogramVec(
+	dialLatencies := prometheus.NewHistogramVec(
 		prometheus.HistogramOpts{
-			Namespace: namespace,
-			Subsystem: subsystem,
-			Name:      "dial_duration_seconds",
+			Namespace: Namespace,
+			Subsystem: Subsystem,
+			Name:      DialLatencyMetric,
```

Comment:
I would inline the metric names, it's confusing to look elsewhere to find the metric name.

Reply:
I want to stay consistent with https://github.com/kubernetes-sigs/apiserver-network-proxy/pull/400, which seemed motivated by unit test reference. @tallclair WDYT

Reply:
All of those metric names should be inlined imo.

Reply:
Ok - I inlined them for client and agent. I kept consistent within server/metrics consistent, since this PR doesn't touch it much.

Reply:
Thanks!